main/src/library/Iterator.flix

```diff
@@ -231,38 +219,30 @@ namespace Iterator {
     ///
     pub def toNec(iter: Iterator[a]): Option[Nec[a]] & Impure =
         let Iterator(done, next) = iter;
-        toNecAcc(done, next, ks -> ks)
-
-    ///
-    /// Helper function for `toNec`.
-    ///
-    def toNecAcc(done: Done, next: Next[a], k: Option[Nec[a]] -> Option[Nec[a]]): Option[Nec[a]] & Impure =
-        if (done()) {
-            k(None)
-        }
-        else {
-            let x = next();
-            toNecAcc(done, next, k1 -> match k1 {
-                case None      => k(Some(Nec.singleton(x)))
-                case Some(nec) => k(Some(Nec.cons(x, nec)))
-            })
-        }
+        def loop(k) = {
```

Comment:
Can we do simpler? Maybe toChain and then toNec? or something?

Reply:
Hmm... it seems to be slightly more complicated than simply adding `toChain` and wrapping that in an `Option` since `Chain` and `Nec` are different data structures (`Nec` is not just a wrapper around `Chain`). To make matters worse, they do not have mutual conversion functions either. I think we should make a new issue that adds this functionality to both data structures which would also make it easier to simplify this.

Reply:
I've tried to simplify it using `foldRight`